Sources/Public/Animation/LottieAnimationView.swift

```diff
@@ -648,15 +648,54 @@ open class LottieAnimationView: LottieAnimationViewBase {
     from dotLottieFile: DotLottieFile)
   {
     guard let dotLottieAnimation = dotLottieFile.animation(for: animationId) else { return }
+    loadAnimation(dotLottieAnimation, from: dotLottieFile)
+  }
 
-    loopMode = dotLottieAnimation.configuration.loopMode
-    animationSpeed = CGFloat(dotLottieAnimation.configuration.speed)
-
-    if let imageProvider = dotLottieAnimation.configuration.imageProvider {
-      self.imageProvider = imageProvider
-    }
+  /// Sets the lottie file backing the animation view. Setting this will clear the
+  /// view's contents, completion blocks and current state. The new animation will
+  /// be loaded up and set to the beginning of its timeline.
+  /// The loopMode, animationSpeed and imageProvider will be set according
+  /// to lottie file settings
+  /// - Parameters:
+  ///   - animationIndex: Internal animation index to play. Optional
+  ///   Defaults to play first animation in file.
+  ///   - dotLottieFile: Lottie file to play
+  public func loadAnimation(
```

Comment:
This new method seems like a reasonable addition to me. As of #2073 you'll want to implement this method on `LottieAnimationLayer` instead of `LottieAnimationView`, and then add a method on `LottieAnimationView` that calls the method on the underlying CALayer.

Reply:
Great, I have moved the function